python/ray/experimental/channel/torch_tensor_nccl_channel.py

```diff
@@ -0,0 +1,222 @@
+import logging
+from typing import TYPE_CHECKING, List, Optional
+
+import ray
+import ray.util.serialization
+from ray.experimental.channel import ChannelContext
+from ray.experimental.channel.common import ChannelInterface
+from ray.experimental.channel.nccl_group import _NcclGroup
+from ray.util.annotations import DeveloperAPI
+
+if TYPE_CHECKING:
+    from ray.experimental.channel.torch_tensor_type import (
+        TorchTensorType,
+        _TorchTensorWrapper,
+    )
+
+try:
+    import torch
+except ImportError:
+    torch = None
+
+
+# Logger for this module. It should be configured at the entry point
+# into the program using Ray. Ray provides a default configuration at
+# entry/init points.
+logger = logging.getLogger(__name__)
+
+
+@DeveloperAPI
+class TorchTensorNcclChannel(ChannelInterface):
+    def __init__(
+        self,
+        writer: ray.actor.ActorHandle,
+        readers: List[ray.actor.ActorHandle],
+        typ: "TorchTensorType",
+        device: Optional["torch.device"] = None,
+    ):
+        from ray.air._internal import torch_utils
+
+        self._nccl_group = None
+
+        self._writer = writer
+        self._writer_rank: Optional[int] = None
+        self._readers = readers
+        self._reader_ranks: Optional[List[int]] = None
+        self._writer_registered: bool = False
+        self._reader_registered: bool = False
+
+        self._device = device
+        if self._device is None:
+            # TODO(swang): Allow default device to be overridden.
+            self._device = torch_utils.get_devices()[0]
+        if self._device.type != "cuda":
+            raise ValueError(
+                f'Actor\'s default device has type "{self._device.type}", need "cuda"'
+            )
+
+        assert typ.transport == "nccl"
+        self._typ = typ
+
+        ctx = ChannelContext.get_current()
+        self._nccl_group = ctx.nccl_group
+        if self._nccl_group is not None:
```

Comment:
when is this None?

Reply:
Hmm good point, I'll change this to assertion.